modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -5254,6 +5374,10 @@ private void processNodeFailedMessage(TcpDiscoveryNodeFailedMessage msg) {
                 return;
             }
 
+            //we will need to recalculate this value since the topology changed
```

Comment:
You must do recalculation only for verified messages!

Reply:
fixed